aiocache/decorators.py

```diff
@@ -268,6 +292,11 @@ class multi_cached:
         ``keys_from_attr``, the decorated callable, and the positional and keyword arguments
         that were passed to the decorated callable. This behavior is necessarily different than
         ``BaseCache.build_key()`` and the call signature differs from ``cached.key_builder``.
+    :param value_checker: dict of callables that allows to control which keys should not
+        be cached based on corresponding value check. The dict keys are set of keys to
+        be checked while the dict value is callable which takes corresponding result value
+        (result[key])  and returns boolean indicating whether cache that (key, value)
+        pair or not.
```

Comment:
Hmm, I'm not sure about this one, it seems to be less flexible than the others.
This requires knowing which keys should (or could) be cached. e.g. You couldn't skip keys that ended with `_nocache`, for example, unless you knew all possible keys beforehand.

I can think of 2 flexible options:
```
skip_keys = skip_cache_func(result.copy())
results_to_cache = {k: v for k, v in result.items() if k not in skip_keys}

# OR

results_to_cache = dict(it for it in result.items() if not skip_cache_func(*it))
```

Reply:
This approach also require you to know all the possible keys.
The original idea was to perform value-based filtering for resulting dict. However I am also not sure this functionality is needed

Reply:
The original idea was to provide an interface to perform value-based filtering (similar to your provided examples). So in both cases you need to know a range of all possible keys. However I m also not sure if this functionality is needed

Reply:
I don't see why you need the possible keys? In the above examples my cache function can look like:
```
def f(results):
    return (k for k in results.keys() if k.endswith("_nocache"))
```
or
```
def f(k, v):
    return k.endswith("_nocache")
```

Whereas the original does not allow filtering keys ending with `_nocache` unless you can name every single possible key beforehand.

Reply:
I'm feeling like the 2nd option is easier and clearer to understand, so I'm leaning towards that solution.

Reply:
Obviously, you can just as easily check values without caring about keys:
```
def f(k, v):
    return v is None
```

Reply:
It seems I ve finally got your idea. Agree -- your variant is more generic and easy-to-use.